cinn/hlir/pe/ir_schedule_pe.cc

```diff
@@ -317,7 +317,11 @@ void IRCudaScheduleReduce(ir::IRSchedule &ir_sch,
   if (parallel_thread_num > max_block_size) {
     auto loops = ir_sch.GetLoops(output->name);
     CHECK_GE(loops.size(), index + 1);
-    for (int idx = 1024; idx > 0; --idx) {
+
+    auto *for_node = loops[index].As<ir::For>();
+    CHECK(for_node->extent.is_constant()) << "The For node's extent must be constant! Please check.";
+    int tot_extent = for_node->extent.get_constant();
```

Comment:
用`ir_schedule_util.h`了的`GetLoopExtent`函数替换这段代码吧
```suggestion
    int tot_extent = GetLoopExtent(loops[index]);
```